### ISSUE TITLE
Clean up crate docs

### ISSUE THREAD

**Issue by brson** [pingcap/fail-rs#27]
The crate docs are pretty overwhelming. Figure out how to defer some of that discussion to elsewhere in the docs.

**Comment by brson** [comment]
I've unassigned myself from this issue. I went over the crate-level docs (which I wrote), and even though there's a lot of it, I like it pretty well, and don't see where better to put that content. I do see that the ["usage in tests"](https://docs.rs/fail/0.2.1/fail/#usage-in-tests) section could be shortened by removing the leading example (that doesn't work), trimming the amount of explanation; but I'm reluctant to because the fail crate is actually pretty tough to use correctly; and also I just like my writing.

I wonder if somebody less attached to these docs could give them an evaluation. Is it ok to release 1.0 with these docs? https://docs.rs/fail/0.2.1/fail/

cc @BusyJay @nrc 

Considering that this is a very small crate, with relatively low usage, I'm inclined to just call it done and bump to 1.0 unless somebody wants to argue for more changes.
